Tangem/Modules/Main/ActionButtonsSwap/ActionButtonsSwap/NotificationManager/ActionButtonsSwapNoficationManager.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  ActionButtonsSwapNoficationManager.swift
+//  TangemApp
+//
+//  Created by GuitarKitty on 29.11.2024.
+//  Copyright © 2024 Tangem AG. All rights reserved.
+//
+
+import Combine
+import Foundation
+
+final class ActionButtonsSwapNotificationManager {
+    private let notificationInputsSubject = CurrentValueSubject<[NotificationViewInput], Never>([])
+
+    private weak var delegate: NotificationTapDelegate?
+
+    private var subscription: AnyCancellable?
+
+    init(
+        statePublisher: AnyPublisher<ActionButtonsSwapViewModel.ActionButtonsTokenSelectorState, Never>
+    ) {
+        subscription = statePublisher
+            .receive(on: DispatchQueue.main)
+            .sink(receiveValue: weakify(
+                self,
+                forFunction: ActionButtonsSwapNotificationManager.setupNotifications(for:)
+            ))
+    }
+
+    private func setupNotifications(
+        for state: ActionButtonsSwapViewModel.ActionButtonsTokenSelectorState
+    ) {
+        var event: ActionButtonsNotificationEvent?
+
+        switch state {
+        case .refreshRequired(let title, let message):
+            event = .refreshRequired(
+                title: title,
+                message: message
+            )
```

Comment:
Nit: но может прям тут просто показывать да и все 
```suggestion
            makeNotification(with: .refreshRequired(
                title: title,
                message: message
            )
            )
```